### ISSUE TITLE
Game crashes on Hypixel Skyblock with `-lwjgl` flag

### ISSUE THREAD

**Issue by chimbersaw** [3arthqu4ke/headlessmc#139]
Hi, love your project!

I have encountered a crash that occurs only when i join Hypixel Skyblock.
Something in this exact world breaks the game.
Without the `-lwjgl` everything works as expected.
Can you please look into it?

Minecraft version: `Fabric 1.19.3`
Mods installed: only `hmc-specifics-1.19.3-fabric.jar`
HeadlessMc version: `headlessmc-launcher-1.9.3.jar`

Steps to reproduce:
* `java -jar headlessmc-launcher-1.9.3.jar`
* `download 1.19.3`
* `fabric 1.19.3`
* `launch fabric-loader-0.15.9-1.19.3 -lwjgl`
* `connect play.hypixel.net`
* `msg /skyblock`

Here's the crash log:
```
---- Minecraft Crash Report ----
// You should try our sister game, Minceraft!

Time: 2024-04-05 10:51:02
Description: Rendering entity in world

java.lang.IllegalStateException: Image is not allocated.
	at net.minecraft.class_1011.method_4320(class_1011.java:153)
	at net.minecraft.class_1011.method_4305(class_1011.java:204)
	at net.minecraft.class_330$class_331.method_1776(class_330.java:95)
	at net.minecraft.class_330$class_331.method_1777(class_330.java:103)
	at net.minecraft.class_330.method_1773(class_330.java:40)
	at net.minecraft.class_915.method_3994(class_915.java:107)
	at net.minecraft.class_915.method_3936(class_915.java:29)
	at net.minecraft.class_898.method_3954(class_898.java:141)
	at net.minecraft.class_761.method_22977(class_761.java:1569)
	at net.minecraft.class_761.method_22710(class_761.java:1316)
	at net.minecraft.class_757.method_3188(class_757.java:1085)
	at net.minecraft.class_757.method_3192(class_757.java:864)
	at net.minecraft.class_310.method_1523(class_310.java:1193)
	at net.minecraft.class_310.method_1514(class_310.java:781)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_1011.method_4320(class_1011.java:153)
	at net.minecraft.class_1011.method_4305(class_1011.java:204)
	at net.minecraft.class_330$class_331.method_1776(class_330.java:95)
	at net.minecraft.class_330$class_331.method_1777(class_330.java:103)
	at net.minecraft.class_330.method_1773(class_330.java:40)
	at net.minecraft.class_915.method_3994(class_915.java:107)
	at net.minecraft.class_915.method_3936(class_915.java:29)
	at net.minecraft.class_898.method_3954(class_898.java:141)
	at net.minecraft.class_761.method_22977(class_761.java:1569)
	at net.minecraft.class_761.method_22710(class_761.java:1316)
	at net.minecraft.class_757.method_3188(class_757.java:1085)

-- Entity being rendered --
Details:
	Entity Type: minecraft:item_frame (net.minecraft.class_1533)
	Entity ID: 39010
	Entity Name: Item Frame
	Entity's Exact location: -0.50, 70.50, -95.97
	Entity's Block location: World: (-1,70,-96), Section: (at 15,6,0 in -1,4,-6; chunk contains blocks -16,0,-96 to -1,255,-81), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at net.minecraft.class_898.method_3954(class_898.java:141)
	at net.minecraft.class_761.method_22977(class_761.java:1569)
	at net.minecraft.class_761.method_22710(class_761.java:1316)
	at net.minecraft.class_757.method_3188(class_757.java:1085)
	at net.minecraft.class_757.method_3192(class_757.java:864)
	at net.minecraft.class_310.method_1523(class_310.java:1193)
	at net.minecraft.class_310.method_1514(class_310.java:781)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Renderer details --
Details:
	Assigned renderer: net.minecraft.class_915@3654474
	Location: 2.00,-1.12,-26.47 - World: (2,-2,-27), Section: (at 2,14,5 in 0,-1,-2; chunk contains blocks 0,0,-32 to 15,255,-17), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Rotation: 0.0
	Delta: 0.4000013
Stacktrace:
	at net.minecraft.class_898.method_3954(class_898.java:141)
	at net.minecraft.class_761.method_22977(class_761.java:1569)
	at net.minecraft.class_761.method_22710(class_761.java:1316)
	at net.minecraft.class_757.method_3188(class_757.java:1085)
	at net.minecraft.class_757.method_3192(class_757.java:864)
	at net.minecraft.class_310.method_1523(class_310.java:1193)
	at net.minecraft.class_310.method_1514(class_310.java:781)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Affected level --
Details:
	All players: 5 total; [class_746['chimbersaw'/38999, l='ClientLevel', x=-2.50, y=70.00, z=-69.50], class_745['RuneXD'/38390, l='ClientLevel', x=-13.84, y=70.00, z=-75.66], class_745['Okapi2211'/33831, l='ClientLevel', x=-28.88, y=70.00, z=-74.53], class_745['ighastly1'/33572, l='ClientLevel', x=-30.03, y=72.00, z=-93.72], class_745['SitToes'/39000, l='ClientLevel', x=-2.50, y=81.00, z=-69.50]]
	Chunk stats: 5041, 60
	Level dimension: minecraft:overworld
	Level spawn location: World: (-3,70,-70), Section: (at 13,6,10 in -1,4,-5; chunk contains blocks -16,0,-80 to -1,255,-65), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Level time: 830957058 game time, 85232 day time
	Server brand: Hypixel BungeeCord (2024.4.1.1) <- vanilla
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:455)
	at net.minecraft.class_310.method_1587(class_310.java:2390)
	at net.minecraft.class_310.method_1514(class_310.java:800)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: vanilla

-- System Details --
Details:
	Minecraft Version: 1.19.3
	Minecraft Version ID: 1.19.3
	Operating System: Mac OS X (aarch64) version 13.4.1
	Java Version: 17.0.9, Homebrew
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Homebrew
	Memory: 2338549184 bytes (2230 MiB) / 3447717888 bytes (3288 MiB) up to 8589934592 bytes (8192 MiB)
	CPUs: 12
	Processor Vendor: Apple Inc.
	Processor Name: Apple M2 Max
	Identifier: Apple Inc. Family 0xda33d83d Model 0 Stepping 0
	Microarchitecture: ARM64 SoC: Avalanche + Blizzard
	Frequency (GHz): 3.70
	Number of physical packages: 1
	Number of physical CPUs: 12
	Number of logical CPUs: 12
	Graphics card #0 name: Apple M2 Max
	Graphics card #0 vendor: Apple (0x106b)
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: unknown
	Graphics card #0 versionInfo: unknown
	Memory slot #0 capacity (MB): 0.00
	Memory slot #0 clockSpeed (GHz): 0.00
	Memory slot #0 type: unknown
	Virtual memory max (MB): 32768.00
	Virtual memory used (MB): 19817.47
	Swap memory total (MB): 0.00
	Swap memory used (MB): 0.00
	JVM Flags: 1 total; -Xss1M
	Launched Version: fabric-loader-0.15.9-1.19.3
	Backend library: LWJGL version 
	Backend API: NO CONTEXT
	Window size: 1920x1080
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Resource Packs: 
	Current Language: English (US)
	CPU: 12x Apple M2 Max
```
[crash-2024-04-05_10.51.02-client.txt](https://github.com/3arthqu4ke/headlessmc/files/14882581/crash-2024-04-05_10.51.02-client.txt)

**Comment by 3arthqu4ke** [comment]
Is that on your Skyblock island only? Or are u in some kind of Skyblock lobby?
I can still attempt to fix this without recreating it.

**Comment by 3arthqu4ke** [comment]
@chimbersaw Could you use the jar published in this workflow: https://github.com/3arthqu4ke/headlessmc/actions/runs/8571329123
And try it with that?

(https://github.com/3arthqu4ke/headlessmc/actions/runs/8571329123/artifacts/1388628556)

**Comment by chimbersaw** [comment]
> Is that on your Skyblock island only? Or are u in some kind of Skyblock lobby? Do you have like images at your skyblock island, like maps? I can still attempt to fix this without recreating it.

This happens when I join Skyblock, so in the main Skyblock hub.

**Comment by chimbersaw** [comment]
> @chimbersaw Could you use the jar published in this workflow: https://github.com/3arthqu4ke/headlessmc/actions/runs/8571329123 And try it with that?
> 
> (https://github.com/3arthqu4ke/headlessmc/actions/runs/8571329123/artifacts/1388628556)

Unfortunately, this didn't fix the problem.

The error reproduces with the steps I mentioned both on my Mac and my ubuntu server.

Please see the recording and the full log:
* Recording: https://streamable.com/3aue3r
* Full log: [full-log.txt](https://github.com/3arthqu4ke/headlessmc/files/14887248/full-log.txt)
* Minecraft crash log: [crash-2024-04-05_17.38.48-client.txt](https://github.com/3arthqu4ke/headlessmc/files/14887251/crash-2024-04-05_17.38.48-client.txt)

**Comment by 3arthqu4ke** [comment]
I made a typo...
Can you try with this one, once it has finished:
https://github.com/3arthqu4ke/headlessmc/actions/runs/8572879296

**Comment by chimbersaw** [comment]
> I made a typo... Can you try with this one, once it has finished: https://github.com/3arthqu4ke/headlessmc/actions/runs/8572879296

It doesn't look good
```
[18:41:49] [Render thread/INFO]: [System] [CHAT]        
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGBUS (0xa) at pc=0x00000001043fafbc, pid=25478, tid=259
#
# JRE version: OpenJDK Runtime Environment Homebrew (17.0.9) (build 17.0.9+0)
# Java VM: OpenJDK 64-Bit Server VM Homebrew (17.0.9+0, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-aarch64)
# Problematic frame:
# V  [libjvm.dylib+0x742fbc]  MarkActivationClosure::do_code_blob(CodeBlob*)+0x3c
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/maxim/Library/Application Support/minecraft/hs_err_pid25478.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/Homebrew/homebrew-core/issues
#
Minecraft exited with code: 134
[18:41:49] [main/INFO] [LaunchCommand]: Deleting b96ac155-5b3e-4f3f-9d03-308006f3439c
```

Error log: [hs_err_pid25478.log](https://github.com/3arthqu4ke/headlessmc/files/14887982/hs_err_pid25478.log)

**Comment by chimbersaw** [comment]
Were you not successful at reproducing this error?

**Comment by 3arthqu4ke** [comment]
Is that a new error?

I also deleted your message, you should sanitize your log files, look for tokens etc..

> Were you not successful at reproducing this error?
I think I will have to play a bit through skyblock to get to the hub?
I think I spawn on the island so I dont crash. Will to play through it sometime.

**Comment by chimbersaw** [comment]
> Is that a new error?
> 
> I also deleted your message, you should sanitize your log files, look for tokens etc..
> 
> > Were you not successful at reproducing this error?
> 
> I think I will have to play a bit through skyblock to get to the hub? I think I spawn on the island so I dont crash. Will to play through it sometime.

Yes, that is a new error. Thanks for removing the comment.

AFAIR it takes no more than 5 minutes to get to the hub (the portal on the neighboring island), and then you start spawning in the hub rather than on your island.

**Comment by chimbersaw** [comment]
Okay i checked that `/hub` works on a new account. So you can just `msg /hub` after `msg /skyblock`

**Comment by 3arthqu4ke** [comment]
I just bridged over ^^

Its odd, I am in the hub, but I do not crash. Could you tell me approximately where I need to go in the hub for the crash to occur?

**Comment by 3arthqu4ke** [comment]
Hmm, but I guess you always spawn at the central place in the hub right?

**Comment by 3arthqu4ke** [comment]
Also, could you just try it one more time, the when an hs_err_pid is created theres a chance that its just a one time thing.

**Comment by chimbersaw** [comment]
> Hmm, but I guess you always spawn at the central place in the hub right?

Yes, you do

> Also, could you just try it one more time, when an hs_err_pid is created theres a chance that its just a one time thing.

It actually worked the second time! However, it crashed with the same message after a few seconds.
The next time there was no crash, the next time the crash occurred a minute or so after joining (i did not move or do anything).

So I've tried this like 5 more times and waited for a couple of minutes each time. In two of those the crash occurred after a few minutes, so idk. Why might this happen?

Here's the cleaned error log: [hs_err_pid26237.log](https://github.com/3arthqu4ke/headlessmc/files/14888940/hs_err_pid26237.log)

You may also want to try to reproduce the error by waiting enough time after `msg /skyblock`

**Comment by 3arthqu4ke** [comment]
But when it crashed on your mac it was always the hs_err_pid, not the `java.lang.IllegalStateException: Image is not allocated.` that you sent me in the beginning, right?

**Comment by 3arthqu4ke** [comment]
Ok whoops, didnt really mean to delete that comment, confused myself.
Comment said basically: hs_err_pid crash occurs still, on mac, but not on ubuntu.

**Comment by 3arthqu4ke** [comment]
Waited until I got marked as AFK and no crash, Windows machine.

**Comment by 3arthqu4ke** [comment]
Are you using any other mods?

**Comment by chimbersaw** [comment]
> But when it crashed on your mac it was always the hs_err_pid, not the `java.lang.IllegalStateException: Image is not allocated.` that you sent me in the beginning, right?
> 
> The `Image is not allocated` should be fixed, I will look into the second one.

* `headlessmc-launcher-1.9.3` - getting `java.lang.IllegalStateException: Image is not allocated` on both platforms
* `headlessmc-launcher-1.9.4` - getting hs_err_pid on mac, no problem on ubuntu

**Comment by 3arthqu4ke** [comment]
Alright thanks. I am going to move that to a new issue, but idk if I can fix it.
I do not have a mac system to recreate the issue and I also do not really understand whats going on in the hs_err_pid.
All I can do is suggest to reinstall java :(
On Windows and Ubuntu it seems to work fine.

**Comment by 3arthqu4ke** [comment]
Potentially related: https://github.com/FTBTeam/FTB-Modpack-Issues/issues/3325
Though there it seems to be Java 17.0.7 only, while this is on 17.0.9?

**Comment by 3arthqu4ke** [comment]
Ok actually might be 17.0.7 onwards, [here](https://github.com/ChosenArchitect/ChosensModdedAdventure/issues/13) it happens on 17.0.8.

@chimbersaw could you downgrade to java 17.0.6 on your MacOS machine?

**Comment by chimbersaw** [comment]
> Potentially related: [FTBTeam/FTB-Modpack-Issues#3325](https://github.com/FTBTeam/FTB-Modpack-Issues/issues/3325) Though there it seems to be Java 17.0.7 only, while this is on 17.0.9?

Thanks for pointing that out, i will try on `17.0.6`

**Comment by chimbersaw** [comment]
> Ok actually might be 17.0.7 onwards, [here](https://github.com/ChosenArchitect/ChosensModdedAdventure/issues/13) it happens on 17.0.8.
> 
> @chimbersaw could you downgrade to java 17.0.6 on your MacOS machine?

Okay, this actually worked: `17.0.6` doesn't crash the game.
I've also tried the latest `17.0.10` release and the crash was present.

**Comment by chimbersaw** [comment]
TL;DR

* `java.lang.IllegalStateException: Image is not allocated` bug is fixed in `headlessmc-1.9.4`
* `hs_err_pid` crash is an aarch64 macOS JVM bug on Java 17.0.7+, consider downgrading to `17.0.6`

Thanks, @3arthqu4ke

**Comment by 3arthqu4ke** [comment]
Nice, thanks for helping out!
